packages/storage/src/indexed-db/index.ts

```diff
@@ -669,4 +674,55 @@ export class IndexedDb implements IndexedDbInterface {
       txs.add({ table: 'SWAPS', value: n.toJson() as Jsonified<SwapRecord> });
     }
   }
+
+  // As more auction types are created, add them to T as a union type.
+  async upsertAuction<T extends DutchAuctionDescription>(
+    auctionId: AuctionId,
+    value: {
+      auction?: T;
+      noteCommitment?: StateCommitment;
+    },
+  ): Promise<void> {
+    const key = uint8ArrayToBase64(auctionId.inner);
+    const existingRecord = await this.db.get('AUCTIONS', key);
+    const auction =
+      (value.auction?.toJson() as Jsonified<T> | undefined) ?? existingRecord?.auction;
+    const noteCommitment =
+      (value.noteCommitment?.toJson() as Jsonified<StateCommitment> | undefined) ??
+      existingRecord?.noteCommitment;
+
+    await this.u.update({
+      table: 'AUCTIONS',
+      key,
+      value: {
+        auction,
+        noteCommitment,
+      },
+    });
+  }
+
+  async *iterateAuctions() {
+    yield* new ReadableStream({
+      start: async controller => {
+        let cursor = await this.db.transaction('AUCTIONS', 'readonly').store.openCursor();
```

Comment:
Can we use for await syntax instead?

```typescript
for await (const cursor of this.db.transaction('AUCTIONS').store) {
          controller.enqueue({
            id: new AuctionId({ inner: base64ToUint8Array(cursor.key) }),
            value: {
              auction: cursor.value.auction
                ? DutchAuctionDescription.fromJson(cursor.value.auction)
                : undefined,
              noteCommitment: cursor.value.noteCommitment
                ? StateCommitment.fromJson(cursor.value.noteCommitment)
                : undefined,
            },
          });
        }
```

Reply:
(Moot point now that I've deleted this method)